### DISCUSSION TITLE
how to integrate ripgrep with fzf by using match positions?

### DISCUSSION THREAD

**Discussion by joelpeapen** [BurntSushi/ripgrep#2580] [Q&A] [answered]
I have this function that does live grep.

```sh
function rgf() {
    c="rg --column -nS --no-heading --color=always"
    f="$(cmd="$c ''" fzf --bind "change:reload:$c {q} || true" \
         --ansi --layout=reverse --preview '' --header 'Search in files' \
         | cut -d':' -f1 | sed 's/\\/\\\\/g;s/"/\\"/g'
    )"
    if [[ -n $f ]]; then
        $EDITOR $f
        cd $(dirname $(readlink -f "$f"))
    fi
}
```

Is there a way to get the line number for the selected file so i can `$EDITOR $f +$linenumber`
to go directly to the line?

**Comment by BurntSushi** [comment]
This looks like an fzf question or a general shell scripting question. The `-n` flag tells ripgrep to show line numbers, which you're already using. Otherwise I'm not familiar enough with fzf to easily answer your question.

**Reply by BurntSushi** [reply]
I also note that you appear to be effectively running `rg ''`, which has got to be just incredibly slow. It also makes clear to me that this isn't a ripgrep question.

**Reply by joelpeapen** [reply]
okay, thanks for pointing out `rg ''`. must have been left over from `man fzf`.

**Reply by joelpeapen** [reply, marked as answer]
found it. maybe this could be in a tips section.

```sh
function rgf() {
    c="rg --column -nS --no-heading --color=always"
    a="$(fzf --bind "change:reload:$c {q} || true" \
        --ansi --reverse --preview '' --header 'Search in files')"
    if [[ -n $a ]]; then
        IFS=':' read -r file line char _ <<< "$a"
        "$EDITOR" "$file" +"$line" -c "norm ${char}lh"
        cd "$(dirname "$(readlink -f "$file")")"
    fi
}
```

**Reply by BurntSushi** [reply]
I personally still have no idea what you're trying to do here, so I wouldn't know how to even describe it.

**Reply by BurntSushi** [reply]
I updated the issue title from a completely genetic thing that meant nothing to me to a more specific guess as to what is being asked here.
